### ISSUE TITLE
Failed to export test result

### ISSUE THREAD

**Issue by dgimb89** [bitrise-steplib/bitrise-step-flutter-test#10]
Hey 👋 ,

since 0.10.0 the step fails on exporting the test results.

This might be related to our project location – the flutter project does not reside in the root directory of the git checkout (~`/Users/vagrant/git/`~ vs `/Users/vagrant/git/flutter/playtestcloud_app`).

```
+------------------------------------------------------------------------------+
| (6) flutter-test                                                             |
+------------------------------------------------------------------------------+
| id: flutter-test                                                             |
| version: 0.10.1                                                              |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2020-07-16T14:11:02Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
Config:
- AdditionalParams: 
- ProjectLocation: /Users/vagrant/git/flutter/playtestcloud_app
- TestResultsDir: /var/folders/6q/wgy6jtp12w5gzgm9lzcglpqw0000gn/T/test_results923061681/test_result345784480
- GenerateCodeCoverageFiles: true
Running test
Command `tojunit` not found, installing...
$ flutter "pub" "global" "activate" "junitreport"
Resolving dependencies...
+ args 1.6.0
+ charcode 1.1.3
+ collection 1.14.13 (1.15.0-nullsafety available)
+ convert 2.1.1
+ intl 0.16.1
+ junitreport 1.3.0
+ meta 1.2.2 (1.3.0-nullsafety available)
+ path 1.7.0
+ petitparser 3.0.4
+ testreport 1.2.0
+ typed_data 1.2.0 (1.3.0-nullsafety available)
+ xml 3.7.0 (4.2.0 available)
Downloading junitreport 1.3.0...
Downloading xml 3.7.0...
Downloading testreport 1.2.0...
Downloading petitparser 3.0.4...
Downloading typed_data 1.2.0...
Downloading meta 1.2.2...
Downloading collection 1.14.13...
Downloading path 1.7.0...
Precompiling executables...
Precompiled junitreport:tojunit.
Installed executable tojunit.
Activated junitreport 1.3.0.
$ flutter "test" "--machine" | tojunit "--output" "./flutter_junit_test_results.xml"
rsync: link_stat "/Users/vagrant/git/./flutter_junit_test_results.xml" failed: No such file or directory (2)
rsync error: some files could not be transferred (code 23) at /BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-52.200.1/rsync/main.c(996) [sender=2.6.9]
Failed to export test result: exit status 23
```

Fixating the `TestResultsDir` to match the `ProjectLocation` does not solve the issue.

**Comment by non-binary** [comment]
Hi @dgimb89,

Thank you for reaching out.

Please send over a build log and enable support access so I can see it.

**Comment by dgimb89** [comment]
Hi @non-binary ,

I enabled support access for the app. Here's the latest build that had the issue: https://app.bitrise.io/build/c3e980f33be54dd9 . I fixed this temporarily by downgrading to 0.9.x for the `flutter-test` step.

Thanks for looking into it!

**Comment by Albert221** [comment]
I have the same issue 👀  Using Flutter Test step version 0.10.1.

**Comment by mjones-droid** [comment]
Hello @dgimb89 ,

This was a little difficult to troubleshoot since one of your tests is currently failing, but I think I figured out what is happening. 

It looks like as of 0.10.x of this step test output is being converted to xml using JUnitReport. I don't believe this was the case for 0.9.x. The error in the original log file was generated when the export to JUnitReport occurred. 

I created a debug workflow and changed the version of the step back to 0.10.x. I was not able to replicate the error. I even double checked and the XML file with the test results was successfully generated. 

I believe the issue was resolved by updating the version of JUnitReport being used. The version used in the log file you sent was 1.3.0. It looks like version 1.3.1 is now being imported. 

I hope that this resolves your issue.
